data/data/agent/files/usr/local/bin/set-node-zero.sh

```diff
@@ -45,6 +45,16 @@ done
 if [ "${IS_NODE_ZERO}" = "true" ]; then
     echo "Node 0 IP ${NODE_ZERO_IP} found on this host" 1>&2
 
+    # Create tls certs if they don't exist. This allows the certs
+    # to be created at run-time, e.g. when installed via the UI
+    AGENT_TLS_DIR=/opt/agent/tls
+    if [ ! -d "$AGENT_TLS_DIR" ]; then
+       mkdir -p $AGENT_TLS_DIR
+       source /usr/local/bin/create-tls-certs.sh
+       cp /tmp/*.key $AGENT_TLS_DIR
+       cp /tmp/*.crt $AGENT_TLS_DIR
+    fi
+
```

Comment:
Given that the agent tls dir is always available ([as per the current code](https://github.com/openshift/installer/blob/c74674db60a35c6960b36fc0df7115ce91acc93b/pkg/asset/agent/image/unconfigured_ignition.go#L202)), I think the overall check could be simplified to something like (I'd remove also the additional script, if not really required):

```
if [ -z $(ls -A "$AGENT_TLS_DIR") ]; then
   IMAGE=$(image_for installer)
   echo "tls certs not found, generating"
   /usr/bin/podman run --privileged -v $AGENT_TLS_DIR:/tmp:Z --rm "${IMAGE}" agent create certificates --dir=/tmp
fi
```

Reply:
After an offline discussion with @bfournie , it looks like simpler to keep the current implmentation, mainly because the openshift-installer command leaves hidden file that should be removed.

/approve

Reply:
Updated.